clients/lsp-javascript.el

```diff
@@ -44,7 +44,7 @@
 
 (defun lsp-typescript-javascript-tsx-jsx-activate-p (filename &optional _)
   "Check if the javascript-typescript language server should be enabled based on FILENAME."
-  (or (string-match-p (rx (one-or-more anything) "." (or "ts" "js") (opt "x") string-end) filename)
+  (or (string-match-p (rx (seq (+ nonl) (submatch (or "mjs" (submatch (any ?j ?t) "s" (? "x")))) string-end)) filename)
```

Comment:
This is more complicated than what I have proposed (which isn't necessary since `rx` is an optimizing regex compiler) and will break for people with newlines in filenames (more of a theoretical point, since lots of other things would probably break at that point; it is possible on Linux, though).

Why not go with my `suggestion`?

Reply:
I've just converted it to a read syntax to match only the extension. It's so simple, there's no need to use rx.